pandas/plotting/_core.py

```diff
@@ -1980,52 +1980,148 @@ def plot_series(data, kind='line', ax=None,                    # Series unique
+        * 'dict' returns a dictionary whose values are the matplotlib
+          Lines of the boxplot.
+        * 'both' returns a namedtuple with the axes and dict.
+        * when grouping with ``by``, a Series mapping columns to
+          ``return_type`` is returned (i.e.
+          ``df.boxplot(column=['Col1','Col2'], by='var',return_type='axes')``
+          may return ``Series([AxesSubplot(..),AxesSubplot(..)],
+          index=['Col1','Col2'])``).
+
+          If ``return_type`` is `None`, a NumPy array
+          of axes with the same shape as ``layout`` is returned
+          (i.e. ``df.boxplot(column=['Col1','Col2'],
+          by='var',return_type=None)`` may return a
+          ``array([<matplotlib.axes._subplots.AxesSubplot object at ..>,
+          <matplotlib.axes._subplots.AxesSubplot object at ..>],
+          dtype=object)``).
+    **kwds : Keyword Arguments (optional)
         All other plotting keyword arguments to be passed to
-        matplotlib's boxplot function
+        `matplotlib.pyplot.boxplot <https://matplotlib.org/api/_as_gen/
+        matplotlib.pyplot.boxplot.html#matplotlib.pyplot.boxplot>`_.
 
     Returns
     -------
-    lines : dict
-    ax : matplotlib Axes
-    (ax, lines): namedtuple
+    result:
+        Options:
+
+        * ax : object of class
+          matplotlib.axes.Axes (for ``return_type='axes'``)
+        * lines : dict (for ``return_type='dict'``)
+        * (ax, lines): namedtuple (for ``return_type='both'``)
+        * :class:`~pandas.Series` (for ``return_type != None``
+          and data grouped with ``by``)
+        * :class:`~numpy.array` (for ``return_type=None``
+          and data grouped with ``by``)
+
+    See Also
+    --------
+    matplotlib.pyplot.boxplot: Make a box and whisker plot.
+    matplotlib.pyplot.hist: Make a hsitogram.
 
     Notes
     -----
     Use ``return_type='dict'`` when you want to tweak the appearance
     of the lines after plotting. In this case a dict containing the Lines
     making up the boxes, caps, fliers, medians, and whiskers is returned.
-    """
 
+    Examples
+    --------
+
+    Boxplots can be created for every column in the dataframe
+    by ``df.boxplot()`` or indicating the columns to be used:
+
+    .. plot::
+        :context: close-figs
+
+        >>> np.random.seed(1234)
+        >>> df = pd.DataFrame(np.random.rand(10,4),
+        ...                   columns=['Col1', 'Col2', 'Col3', 'Col4'])
+        >>> boxplot = df.boxplot(column=['Col1', 'Col2', 'Col3'])
+
+    Boxplots of variables distributions grouped by a third variable values
+    can be created using the option ``by``. For instance:
+
+    .. plot::
+        :context: close-figs
+
+        >>> df = pd.DataFrame(np.random.rand(10,2), columns=['Col1', 'Col2'] )
+        >>> df['X'] = pd.Series(['A','A','A','A','A','B','B','B','B','B'])
+        >>> boxplot = df.boxplot(by='X')
+
+    A list of strings (i.e. ``['X','Y']``) containing can be passed to boxplot
```

Comment:
This sentence doesn't sound correct, it's missing some noun at "a list of strings containing (what?) can be passed to boxplot".